### PR TITLE
Upgrade Kotlin and Arrow

### LINKED ISSUE #9: Upgrade Kotlin and Arrow
* Update Kotlin to `1.2.61` to match Kotlin version used by lastest Arrow release.
* Update Arrow to `0.7.3`.

### PR DESCRIPTION
**Pull request by JorgeCastilloPrz**
### :pushpin: References
**Issues:**
* Fixes #9 

### :tophat: What is the goal?

* Update Kotlin to `1.2.61` to match Kotlin version used by lastest Arrow release.
* Update Arrow to `0.7.3`.

### 💻 How is it being implemented?

* Upgraded both versions.
* Adapter the whole project to use the lastest Arrow released syntax.

There's no intention on refactoring or redesigning this yet. That'll need to wait a bit more. This is mostly a middle step to start working on that when it's possible. And also for me to get to understand how `Kollect` works.

### 📱 How to Test

* No much to test yet. Better just take a look to what has been ported and report anything found that doesn't make much sense regarding what the library had implemented before these changes.

### CODE REVIEW

**Comment by raulraja** [comment]
Thanks @JorgeCastilloPrz this brings it to a good starting point to redo Kollect in terms of `arrow-effect` like it's being done in its scala counterpart https://github.com/47deg/fetch/pull/155 https://github.com/47deg/fetch/pull/156

**Comment by JorgeCastilloPrz** [comment]
Yep @raulraja I've been talking to Fede about this, he warned me about the incoming redesign that will follow the one in Fetch being done by Alejandro.

File: kollect-core/src/main/kotlin/kollect/arrow/instances/core/instances.kt [hunk 1]
```diff
@@ -1,16 +1,19 @@
...
-import arrow.typeclasses.MonadError
+import arrow.instance
+import arrow.instances.TryMonadInstance
 import kollect.KollectMonadError
 import kollect.Query
 
-object TryKollectMonadErrorInstance : KollectMonadError<ForTry> {
-    override fun ME(): MonadError<ForTry, Throwable> = Try.monadError()
+@instance(Try::class)
+interface TryKollectMonadErrorInstance : TryMonadInstance, KollectMonadError<ForTry> {
```
Comment (pakoito) [inline kollect-core/src/main/kotlin/kollect/arrow/instances/core/instances.kt]: TryMonadErrorInstance?

Comment (raulraja) [reply kollect-core/src/main/kotlin/kollect/arrow/instances/core/instances.kt]: @pakoito the issue here is that the Monad here is a Free monad and uses suspension so it required a monad error that was custom but once this code is moved to arrow effect this type class and instance won't be necessary

**Pull request merged**